index.js

```diff
@@ -18,49 +44,58 @@ module.exports = {
     this.emberVersion = checker.forEmber();
   },
 
-  config() {
-    // do nothing if running with Ember 2.x
-    if (this.emberVersion.lt(minEmberVersion)) {
-      return;
-    }
-
-    return this._super.config.apply(this, arguments);
-  },
-
   included() {
     this._super.included.apply(this, arguments);
 
+    const config = this.project.configCache;
+    const env = this.app.env;
+
+    // This configures EmberENV in a "hacky" way as the 'config' hook is called
+    // before the options from EmberApp are provided. There currently is no way of handling
+    // this case other than manually messing with configCache or modifing the dom by setting
+    // up window.EmberENV before ember loads.
+    if (!config.get(env)) {
```

Comment:
@rwjblue Can you take a look at this. This is the hacky bit

Reply:
Instead of this, can we use `this.project.config(env)` to prime the cache?